iroh-rpc/src/stream.rs

```diff
@@ -1,96 +1,190 @@
 }
 
-impl Stream for InStream {
+impl Stream for OrderedStream {
     type Item = Result<Vec<u8>, RpcError>;
     fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
-        // check if the stream has finished its task successfully
+        // check if the stream has already finished its task successfully
         let num_chunks = self.num_chunks;
         let next_chunk = self.next_chunk;
         if next_chunk == num_chunks {
-            debug!(target: "InStream", "No more items in stream.");
+            debug!(target: "OrderedStream", "No more items in stream.");
+
             return Poll::Ready(None);
         }
+
+        // check if there are errors we should be returning
+        if let Ok(Some(e)) = self.error_receiver.try_next() {
+            error!(target = "OrderedStream", "stream error {}", e);
+            return Poll::Ready(Some(Err(e)));
+        }
+
         // check if the chunk we are looking for is a chunk we already have
-        if let Some(c) = self.chunks.remove(&next_chunk) {
-            debug!(target: "InStream", "Returning already received chunk");
+        let chunks = Arc::clone(&self.chunks);
+        let mut chunks = chunks.lock().unwrap();
+        if let Some(c) = chunks.remove(&next_chunk) {
+            debug!(target: "OrderedStream", "Returning an already received chunk");
             self.next_chunk += 1;
             return Poll::Ready(Some(Ok(c)));
         }
-        // otherwise, check the channel to see if we can get the next chunk
-        match self.packet_receiver.poll_next_unpin(cx) {
-            Poll::Pending => Poll::Pending,
-            Poll::Ready(res) => match res {
-                None => Poll::Ready(Some(Err(RpcError::StreamClosed))),
-                Some(p) => match p {
-                    StreamType::Packet(p) => {
-                        if next_chunk == p.index {
-                            self.next_chunk += 1;
-                            debug!(target: "InStream", "Returning just received chunk {}", p.index);
-                            return Poll::Ready(Some(Ok(p.data)));
-                        }
-                        debug!(target: "InStream", "Storing out of order chunk {}", p.index);
-                        self.chunks.insert(p.index, p.data);
-                        Poll::Pending
-                    }
-                    StreamType::RpcError(e) => {
-                        error!(target: "InStream", "Received error off of the stream: {:?}", e);
-                        Poll::Ready(Some(Err(e)))
-                    }
-                },
-            },
-        }
-    }
-}
-
-impl Drop for InStream {
-    fn drop(&mut self) {
-        self.out_sender
-            .try_send(Command::CloseStream { id: self.id })
-            .expect("Out sender to still be active.");
+        // otherwise, we don't have the chunk, so let's wait until the next
+        // message
+        debug!(
+            target = "OrderedStream",
```

Comment:
why do you need to add this `target = `?

Reply:
I guess I don't, I just grouped all the stuff from the streaming mod so I could isolate the logs when I was debugging.

It's all throughout this file, should I remove the targets?

Reply:
it's a bit cleaner without, given that we don't use it anywhere else and we already capture the location automatically

Reply:
Got it, I'll remove it!